reference/test/matrix/ell_kernels.cpp

```diff
@@ -0,0 +1,189 @@
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <core/matrix/ell.hpp>
+
+
+#include <gtest/gtest.h>
+
+
+#include <core/base/exception.hpp>
+#include <core/base/exception_helpers.hpp>
+#include <core/base/executor.hpp>
+#include <core/matrix/dense.hpp>
+#include <core/test/utils/assertions.hpp>
+
+
+namespace {
+
+
+class Ell : public ::testing::Test {
+protected:
+    using Mtx = gko::matrix::Ell<>;
+    using Vec = gko::matrix::Dense<>;
+
+    Ell()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx(Mtx::create(exec, 2, 3, 3, 2))
+    {
+        Mtx::value_type *v = mtx->get_values();
+        Mtx::index_type *c = mtx->get_col_idxs();
+        Mtx::index_type n = mtx->get_max_nonzeros_per_row();
+        n = 3;
+        c[0] = 0;
+        c[1] = 1;
+        c[2] = 1;
+        c[3] = 0;
+        c[4] = 2;
+        c[5] = 0;
+        v[0] = 1.0;
+        v[1] = 5.0;
+        v[2] = 3.0;
+        v[3] = 0.0;
+        v[4] = 2.0;
+        v[5] = 0.0;
+    }
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::unique_ptr<Mtx> mtx;
+};
+
+
+TEST_F(Ell, AppliesToDenseVector)
+{
+    auto x = gko::initialize<Vec>({2.0, 1.0, 4.0}, exec);
+    auto y = Vec::create(exec, 2, 1, 1);
+
+    mtx->apply(x.get(), y.get());
+
+    EXPECT_EQ(y->at(0), 13.0);
+    EXPECT_EQ(y->at(1), 5.0);
+}
+
+
+TEST_F(Ell, AppliesToDenseMatrix)
+{
+    auto x = gko::initialize<Vec>({{2.0, 3.0}, {1.0, -1.5}, {4.0, 2.5}}, exec);
+    auto y = Vec::create(exec, 2, 2, 2);
+
+    mtx->apply(x.get(), y.get());
+
+    EXPECT_EQ(y->at(0, 0), 13.0);
+    EXPECT_EQ(y->at(1, 0), 5.0);
+    EXPECT_EQ(y->at(0, 1), 3.5);
+    EXPECT_EQ(y->at(1, 1), -7.5);
+}
+
+
+TEST_F(Ell, AppliesLinearCombinationToDenseVector)
+{
+    auto alpha = gko::initialize<Vec>({-1.0}, exec);
+    auto beta = gko::initialize<Vec>({2.0}, exec);
+    auto x = gko::initialize<Vec>({2.0, 1.0, 4.0}, exec);
+    auto y = gko::initialize<Vec>({1.0, 2.0}, exec);
+
+    mtx->apply(alpha.get(), x.get(), beta.get(), y.get());
+
+    EXPECT_EQ(y->at(0), -11.0);
+    EXPECT_EQ(y->at(1), -1.0);
+}
+
+
+TEST_F(Ell, AppliesLinearCombinationToDenseMatrix)
+{
+    auto alpha = gko::initialize<Vec>({-1.0}, exec);
+    auto beta = gko::initialize<Vec>({2.0}, exec);
+    auto x = gko::initialize<Vec>({{2.0, 3.0}, {1.0, -1.5}, {4.0, 2.5}}, exec);
```

Comment:
It would be nice to have this formatted like:
`// clang-format off
auto mtx2 = gko::initialize<gko::matrix::Csr<>>(
        {{1.0, 3.0, 2.0},
         {0.0, 5.0, 0.0},
         {0.0, 1.5, 2.0}}, exec);
// clang-format on
`
THis way, it is easier for the reader to grasp what is initialized here.

Reply:
Fix it.